website/docs/guides/migration/versions/upgrading-to-dbt-utils-v1.0.md

```diff
@@ -0,0 +1,116 @@
+
+Just like the switch to dbt Core 1.0 last year, there are some breaking changes as we standardised and prepared for the future. Most changes can be handled with find-and-replace; if you need help, post on the [Community Forum](https://discourse.getdbt.com) or in #package-ecosystem on Slack. 
+
+## New features
+
+- `get_single_value()` is an easy way to pull a single value from a SQL query, instead of having to access the `[0][0]`th element of a `run_query` result.
+- `safe_divide()` returns null when the denominator is 0, instead of throwing a divide-by-zero error.
+- The new `not_empty_string` test is an easier wrapper than using `expression_is_true` to check the length of a column.
+
+## Enhancements
+
+- Many tests are more meaningful when run against subgroups of a table, e.g. validating that recent data exists for every turnstile. Add the new `group_by_columns` argument to your tests to do so. Check out [https://www.emilyriederer.com/post/grouping-data-quality/](https://www.emilyriederer.com/post/grouping-data-quality/) for more information.
+- With the addition of an on-by-default `quote_identifiers` flag in the `star()` macro, you can now disable quoting if necessary.
+
+## Fixes
+
+- `union()` now includes/excludes columns case-insensitively
+- `slugify()` prefixes an underscore when the first char is a digit
+- The `expression_is_true` test doesn’t output `*` unless storing failures, a cost improvement for BigQuery.
+
+## Breaking Changes
+
+- `surrogate_key()` has been replaced by `generate_surrogate_key()`. The original treated null values and blank strings the same, which could lead to duplicate keys being created.  `generate_surrogate_key()` does not have this flaw. Compare the [surrogate keys calculated for these columns](https://docs.google.com/spreadsheets/d/1qWfdbieUOSgkzdY0kmJ9iCgdqyWccA0R-6EW0EgaMQc/edit#gid=0):
+
+![A table comparing the behaviour of surrogate_key and generate_surrogate_key](/img/guides/migration/versions/surrogate_key_behaviour.png)
+
+Changing the calculation method for surrogate keys, even for the better, could have significant consequences in downstream uses (such as snapshots and incremental models which use this column as their `unique_key`). Because of this, it is possible to opt in to the legacy behaviour by setting the following variable in your dbt project:
+
+```yaml
+#dbt_project.yml
+vars:
+  surrogate_key_treat_nulls_as_empty_strings: true #turn on legacy behaviour
+```
+
+By creating a new macro instead of updating the behaviour of the old one, we are requiring all projects who use this macro to make an explicit decision about which approach is better for their context. **Our recommendation is that existing users should opt into the legacy behaviour** unless you are confident that either a) your surrogate keys never contained nulls, or b) your surrogate keys are not used for incremental models, snapshots or other stateful artifacts and so can be regenerated with new values without issue.
```

Comment:
```suggestion
By creating a new macro instead of updating the behavior of the old one, we require all projects which use this macro to make an explicit decision about which approach is better for their context. 

**Our recommendation is that existing users should opt into the legacy behavior** unless you are confident that either:

a) your surrogate keys never contained nulls, or 
b) your surrogate keys are not used for incremental models, snapshots, or other stateful artifacts and can be regenerated with new values without issue.
```